### DISCUSSION TITLE
Websockets with OpenIDC

### DISCUSSION THREAD

**Discussion by Zowlyfon** [OpenIDC/mod_auth_openidc#1052] [Q&A]
Hi,

We have an SSO proxy setup using mod_auth_openidc and session cookies, a web application we're developing requires a websocket connection, and ideally we'd like to have this behind the SSO proxy aswell, however the browser websocket API doesn't allow the addition of headers to the request meaning that it's not authenticating with mod_auth_openidc.

Does anyone know of any way to get mod_auth_openidc to authenticate a browser websocket connection?

Thank you for reading.

**Comment by zandbelt** [comment]
I believe there are some Discussions in this project that suggest protecting the app with AuthType openid-connect and then use ProxyPass to proxy to websockets; I don't think headers are an issue here unless you need to propagate information over the websocket
